docs/DeveloperGuide.md

```diff
@@ -173,6 +173,96 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+<puml src="diagrams/AddTagActivityDiagram.puml", width="250"></puml>
+
+#### Design considerations:
+
+**Aspect: How AddTagCommand executes:**
+* **Alternative 1 (current choice):** Builts a new Person.
+    * Pros: Easy to implement (using `EditCommand` as reference), immutability allows for potential redo and undo commands.
+    * Cons: Memory intensive, costly in terms of time.
+* **Alternative 2:** Add tags to `Person`.
+    * Pros: Memory efficient
+    * Cons: Mutable `Person` can affect implementation of potential redo and undo commands.
+
+### The Leave class
+
+ManageHR keeps track of the leaves of employees within the company with the use of `Leave` and `UniqueLeaveList`. The `UniqueLeaveList` serves as a container for the `Leave` objects,
+while enforcing the constraints that no 2 leaves can have same start date and end date for the same employee.
+
+The `Leave` class contains the following attributes.
+<puml src="diagrams/LeaveObjectDiagram.puml", width="250"></puml>>
+
+1. `ComparablePerson`: The employee.
+2. `Title`: The title of the leave.
+3. `Description`: The description of the leave.
+4. `Date start`: The start date of the leave.
+5. `Date end`: The end date of the leave.
+5. `Status`: The status of the leave.
+
+`Date start` must be earlier than or the same as the `Date end`.
+All the attributes except Description and Status are compulsory fields for adding a leave.
+
+### Add A Leave application feature
+
+The add-leave command allows the HR manager to add a leave record for a specific employee. This feature enhances the HRMate system by providing a way to manage and track employee leaves efficiently.
+Fields compulsory to enter for `add-leave` as `String` type include:
+1. `index`: The index of the person for the leave application, it will be parsed as `Index` type to `AddLeaveCommand` object and converted to `Person` type based on the displayed list when a new `Leave` object created.
+2. `title`: The title of the leave, it will be parsed as `Title` type to `AddLeaveCommand` object and to the newly created `Leave` object.
+3. `date start`: The start date of the leave in "yyyy-MM-dd" format, it will be parsed together with `date end` as `Range` type to `AddLeaveCommand` object and to the newly created `Leave` object.
+4. `date end`: The end date of the leave in "yyyy-MM-dd" format, it will be parsed together with `date start` as `Range` type to `AddLeaveCommand` object and to the newly created `Leave` object.
+5. `description`: The description of the leave, it is optional, it will be parsed as `NONE` if no description field exists, otherwise it will be parsed as `Description` type to `AddLeaveCommand` object and to the newly created `Leave` object.
+
+* CommandException will be thrown at `AddLeaveCommand` for unfounded index in the list.
+* The rest of the invalid fields exception including illegal arguments and end date before start date will be handled by ParserException in `AddLeaveCommandParser`.
```

Comment:
Can rephrase this as "InvalidValueExceptions thrown due to illegal arguments, and EndBeforeStartExceptions thrown due to the end date occurring before the start date, will be handled by ParserException in `AddLeaveCommandParser`